helper/temporal/temporal.go

```diff
@@ -0,0 +1,25 @@
+package temporal
+
+import (
+	"gopkg.in/src-d/go-billy.v3"
+	"gopkg.in/src-d/go-billy.v3/util"
+)
+
+// Temporal is a helper that implements billy.TempFile over any filesystem.
+type Temporal struct {
+	billy.Filesystem
+	tmpdir string
+}
+
+// New creates a new filesystem wrapping up 'fs' the intercepts the calls to
+// the TempFile method.
```

Comment:
The second argument to this function needs to be explained.  Especially how does it modifies the logic of the `TempFile` method first argument.